src/history/hs_rec.c

```diff
@@ -327,8 +326,13 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_MULTI *mult
     insert_cnt = 0;
     WT_TIME_WINDOW_INIT(&tw);
 
-    WT_RET(__wt_curhs_open(session, NULL, &hs_cursor));
-    F_SET(hs_cursor, WT_CURSTD_HS_READ_COMMITTED);
+    if (r->hs_cursor == NULL)
+        WT_RET(__wt_curhs_open(session, NULL, &r->hs_cursor));
+
+    cur_hs_read_committed = F_ISSET(r->hs_cursor, WT_CURSTD_HS_READ_COMMITTED);
+
+    if (!cur_hs_read_committed)
```

Comment:
This check is redundant, we want the flag set anyway.

Reply:
This change became obsolete after 4a15e69